delta/models/text_match_model.py

```diff
@@ -114,3 +115,162 @@ def call(self, inputs, training=None, mask=None):  # pylint: disable=too-many-lo
     scores = self.final_dense(out)
 
     return scores
+
+
+# pylint: disable=too-many-instance-attributes,too-many-ancestors
+@registers.model.register
+class MatchPyramidTextClassModel(MatchRnn):
+  """Match texts model with Match Pyramid."""
+
+  def __init__(self, config, **kwargs):
+    super().__init__(config, **kwargs)
+    logging.info("Initialize MatchPyramidTextClassModel ...")
+
+    self.vocab_size = config['data']['vocab_size']
+    self.num_classes = config['data']['task']['classes']['num_classes']
+    self.max_seq_len = config['data']['task']['max_seq_len']
+    model_config = config['model']['net']['structure']
+    self.dropout_rate = model_config['dropout_rate']
+    self.embedding_size = model_config['embedding_size']
+    self.emb_trainable = model_config['emb_trainable']
+    self.lstm_num_units = model_config['lstm_num_units']
+    self.fc_num_units = model_config['fc_num_units']
+    self.l2_reg_lambda = model_config['l2_reg_lambda']
+
+    # Number of convolution blocks
+    self.num_blocks = model_config['num_blocks']
+    # The kernel count of the 2D convolution
+    self.kernel_count = model_config['kernel_count']
+    # The kernel size of the 2D convolution of each block
+    self.kernel_size = model_config['kernel_size']
+    # The max-pooling size of each block
+    self.dpool_size = model_config['dpool_size']
+    # The padding mode in the convolution layer
+    self.padding = model_config['padding']
+    # The activation function
+    self.activation = model_config['activation']
+
+    self.embed = tf.keras.layers.Embedding(
+      self.vocab_size,
+      self.embedding_size,
+      trainable=self.emb_trainable,
+      name='embdding',
+      embeddings_initializer=self.embed_initializer)
+
+    self.embed_d = tf.keras.layers.Dropout(self.dropout_rate)
+
+    # TODO
```

Comment:
Is here anything left TODO?

Reply:
Do you mean add `matching_type` into configure file?

Reply:
Fixed